### PR TITLE
Update Getting started on Linux to include git cloning information.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
Update getting_started_on_linux.md to add git configuration snippet similar to getting_started_on_windows.md

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
Go ahead and ignore the Kokoro failure. I'm investigating, but it's nothing to do with your change

File: docs/getting_started_on_linux.md [hunk 1]
```diff
@@ -68,6 +68,42 @@ environment variable is set when you are building. You may find it useful to add
 `source [PATH TO VULKAN SDK]/setup-env.sh` to your `~/.bashrc` file to simplify
 environment variable setup.
 
+## Optional: Configure Git
+
+### Git SSH
```
Comment (GMNGeoffrey) [inline docs/getting_started_on_linux.md]: I would just link the GitHub docs on this. There are so many key configuration options: https://help.github.com/en/enterprise/2.15/user/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

Oh I see this mirrors the windows instructions

File: docs/getting_started_on_linux.md [hunk 2]
```diff
@@ -68,6 +68,42 @@ environment variable is set when you are building. You may find it useful to add
...
+`IREE_CLONE_ARGS="--reference=/path/to/reference/repo"`. See
+`build_tools/scripts/populate_reference_repo.sh` for further details.
+
+```shell
+IREE_CLONE_ARGS=""
+mkdir -p /home/user/ireepub
+cd /home/user/ireepub
+git clone $IREE_CLONE_ARGS https://github.com/google/iree.git iree
+cd iree
+git submodule init
```
Comment (GMNGeoffrey) [inline docs/getting_started_on_linux.md]: Is --recursive necessary? I don't think I've ever used it

File: docs/getting_started_on_linux.md [hunk 3]
```diff
@@ -68,6 +68,42 @@ environment variable is set when you are building. You may find it useful to add
...
+Note that if you will be cloning frequently, it can be sped up significantly by
+creating a reference repo and setting
+`IREE_CLONE_ARGS="--reference=/path/to/reference/repo"`. See
+`build_tools/scripts/populate_reference_repo.sh` for further details.
+
+```shell
+IREE_CLONE_ARGS=""
+mkdir -p /home/user/ireepub
+cd /home/user/ireepub
+git clone $IREE_CLONE_ARGS https://github.com/google/iree.git iree
```
Comment (GMNGeoffrey) [inline docs/getting_started_on_linux.md]: Naming it "iree" explicitly is fine, but it's also the default name, so unnecessary

File: docs/getting_started_on_linux.md [hunk 4]
```diff
@@ -68,6 +68,42 @@ environment variable is set when you are building. You may find it useful to add
...
+### Other git config options
+
+```shell
+git config --global user.name "MY NAME"
+git config --global user.email "MY EMAIL"
+```
+
+### Clone
+
+This assumes that we are cloning into `/home/user/ireepub`. Update accordingly for
```
Comment (GMNGeoffrey) [inline docs/getting_started_on_linux.md]: ireepub seems like an unlikely directory name for an external contributor to use and "user" is unlikely to be their username. What about using an env variable that references "$HOME".

Also I think I would say that we're cloning *into* `/home/user/ireepub/iree` in this case. What about "cloning under"

```suggestion
This assumes that we are cloning under `$HOME/git`. Update accordingly for
```

Looks like there's no way to do multi-line suggested changes (what's even the point then?), but that would make the below shell block


```shell
BASE_DIR="${HOME?}/git"
mkdir -p ${BASE_DIR?}
cd ${BASE_DIR?}
git clone $IREE_CLONE_ARGS https://github.com/google/iree.git iree
cd iree
git submodule init
git submodule update $IREE_CLONE_ARGS --recursive
```

File: docs/getting_started_on_windows.md [hunk 5]
```diff
@@ -134,7 +134,7 @@ environment variable is set when you are building.
 
 ```shell
 # Disable stupid^H^H^H^H^H^H Windows line ending translation
-git config --global core.autocrlf false
+git config --global core.autocrlf true
```
Comment (GMNGeoffrey) [inline docs/getting_started_on_windows.md]: Personally, I think this should really be a separate PR with an explanation of what's going on here, but I guess Ben did ask you to throw this in and merging via copybara is still pretty painful, which works against small PRs :-/

Ah well

File: docs/getting_started_on_linux.md [hunk 6]
```diff
@@ -68,6 +68,42 @@ environment variable is set when you are building. You may find it useful to add
 `source [PATH TO VULKAN SDK]/setup-env.sh` to your `~/.bashrc` file to simplify
 environment variable setup.
 
+## Optional: Configure Git
```
Comment (GMNGeoffrey) [inline docs/getting_started_on_linux.md]: It sounds a bit weird to call this "optional" since you definitely need git for everything :-D

I see this mirrors the windows instructions.

File: docs/getting_started_on_linux.md [hunk 7]
```diff
@@ -68,6 +68,42 @@ environment variable is set when you are building. You may find it useful to add
...
+*   Try a test connection `ssh git@github.com`
+
+### Other git config options
+
+```shell
+git config --global user.name "MY NAME"
+git config --global user.email "MY EMAIL"
+```
+
+### Clone
```
Comment (GMNGeoffrey) [inline docs/getting_started_on_linux.md]: Let's put this one level up. Cloning isn't part of configuring git.

```suggestion
## Clone
```

**Comment by GMNGeoffrey** [comment]
Kokoro mystery solved. Your branch is rebased 10 commits back on master, so the presubmit build configuration doesn't exist yet.

File: docs/getting_started_on_linux.md (hunk 1 above)
Comment (MaheshRavishankar) [reply docs/getting_started_on_linux.md]: Yeah. I just copy pasted it

File: docs/getting_started_on_linux.md (hunk 6 above)
Comment (MaheshRavishankar) [reply docs/getting_started_on_linux.md]: Should I leave this as is then?

File: docs/getting_started_on_linux.md (hunk 7 above)
Comment (MaheshRavishankar) [reply docs/getting_started_on_linux.md]: Ok. Will change it

File: docs/getting_started_on_linux.md (hunk 4 above)
Comment (MaheshRavishankar) [reply docs/getting_started_on_linux.md]: Why is base_dir "${HOME?}/git"

Comment (MaheshRavishankar) [reply docs/getting_started_on_linux.md]: `ireepub` is what is used on windows side.

File: docs/getting_started_on_linux.md (hunk 3 above)
Comment (MaheshRavishankar) [reply docs/getting_started_on_linux.md]: Keeping it consistent with the Windows side.

File: docs/getting_started_on_linux.md (hunk 2 above)
Comment (MaheshRavishankar) [reply docs/getting_started_on_linux.md]: Maybe better to use --recursive.

File: docs/getting_started_on_linux.md (hunk 4 above)
Comment (GMNGeoffrey) [reply docs/getting_started_on_linux.md]: Oh huh, weird... Ok let's keep it consistent then. Not sure I understand the question, but it's just an arbitrary subdirectory under home (~). It happens to be where I stick my git repositories. I don't actually care what we put here (what do most people do?), but I think we should use `$HOME` instead of `/home/user` just to make copy-pasting this a bit easier.
